### PR TITLE
Find and replace all %>% with native pipe (|>)

### PR DESCRIPTION
**Pull request by e-perl-NOAA**
See issue #916.

### CODE REVIEW

**Comment by iantaylor-NOAA** [comment]
I'd like to get more feedback on whether folks are using r4ss with older versions of R before merging this. I'll send a message via the ss3-forum.

Note that we also need to change the minimum version of R on this line: https://github.com/r4ss/r4ss/blob/f2de4576bbca1512b8ce1d47227bdda213f6e3d0/DESCRIPTION#L58 as discussed in #916.

**Review by iantaylor-NOAA — APPROVED** [review]
I haven't heard any concerns about the switch to R 4.1.0 so let's go ahead and merge this.

**Comment by e-perl-NOAA** [comment]
@iantaylor-NOAA the run-call-extra-tests-for-code-coverage failed last time so I want to make sure it passes this time before merging.

**Comment by iantaylor-NOAA** [comment]
@e-perl-NOAA, good catch. I wasn't paying attention to the tests for this PR. 
I find the errors at https://github.com/r4ss/r4ss/actions/runs/10010441623/job/27671624071#step:11:205 to be very confusing and don't see how they would be related to the changes you made.
I'll try running some pieces of the tests locally to see if I see anything going wrong.

File: R/SS_output.R [hunk 1]
```diff
@@ -2278,31 +2278,31 @@ SS_output <-
...
+              dplyr::rename(Like_sum = Like) |> # like for vector not bin
+              dplyr::select(Fleet, Time, Sexes, Part, Nsamp_DM) |>
               dplyr::left_join(lendbase, .)
           }
           # add info to age comp data
           if (nrow(agedbase) > 0) {
-            agedbase <- fit_age_comps %>%
-              dplyr::rename(Like_sum = Like) %>% # like for vector not bin
-              dplyr::select(Fleet, Time, Sexes, Part, Nsamp_DM) %>%
+            agedbase <- fit_age_comps |>
```
[the remark this answers is not in the document]
Comment (e-perl-NOAA) [reply R/SS_output.R:2288]: Okay, based on reading this, it seems like the base pipe can't replace all instances of the magrittr pipe so I might just have to revert here.

[the remark this answers is not in the document]
Comment (e-perl-NOAA) [reply R/SS_output.R:2288]: Or at least in the places before the (.) notation it seems.

**Comment by e-perl-NOAA** [comment]
@iantaylor-NOAA I had to re-write those series of functions since native pipe doesn't work well with dot notation. It looks like it's working now that I've re-written those series of functions.

**Comment by e-perl-NOAA** [comment]
I just added some lines in the description file to try to parallelize the tests and make it so that it doesn't take 4+ hours for the codecov test to complete.

**Comment by iantaylor-NOAA** [comment]
@e-perl-NOAA, figuring out how to speed up the extra tests can wait for a separate pull request.
Thanks for getting this issue dealt with.

**Pull request merged**